### PR TITLE
Port window-new-total to Rust.

### LINKED ISSUE #873: Port window-new-total
```c
DEFUN ("window-new-total", Fwindow_new_total, Swindow_new_total, 0, 1, 0,
       doc: /* Return the new total size of window WINDOW.
WINDOW must be a valid window and defaults to the selected one.

The new total size of WINDOW is the value set by the last call of
`set-window-new-total' for WINDOW.  If it is valid, it will be shortly
installed as WINDOW's total height (see `window-total-height') or total
width (see `window-total-width').  */)
  (Lisp_Object window)
{
  return decode_valid_window (window)->new_total;
}
```

### PR DESCRIPTION
**Pull request by AshtonKem**
Fixes #873.

### CODE REVIEW

File: test/rust_src/src/windows-tests.el [hunk 1]
```diff
@@ -38,6 +38,10 @@
   (should (eq (window-old-point) 1))
   (should (eq (window-old-point (selected-window)) 1)))
```
Comment (shaleh) [inline test/rust_src/src/windows-tests.el]: Awesome. Tests rock.

Why 9? Can we use a variable or at a minimum have a comment documenting the choice

**Comment by AshtonKem** [comment]
I set it to 1 and ran it, and the test failed because the correct value was 9. I assume that there might be some minimum buffer height value somewhere, if I can find it I’ll reference it in the test.

File: test/rust_src/src/windows-tests.el (hunk 1 above)
Comment (AshtonKem) [reply test/rust_src/src/windows-tests.el]: That indeed does work, and is at least a little less magical than hard-coding a 9. It looks like the relationship between new-total and total-height is a bit more complicated than this call would imply, but that's probably well out of the scope of this MR.

Comment (shaleh) [reply test/rust_src/src/windows-tests.el]: Agreed.

**Pull request merged**